controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -489,6 +468,38 @@ func (r *StorageConsumerReconciler) reconcileConsumerConfigMap(
 				return err
 			}
 		}
+
+		if controllerRef != nil && controllerRef.UID == r.storageConsumer.UID {
+			if consumerConfigMap.Data == nil {
+				consumerConfigMap.Data = map[string]string{}
+			}
+			defaultConsumerResourceNames := util.GetStorageConsumerDefaultResourceNames(
+				r.storageConsumer.Name,
+				string(r.storageConsumer.UID),
+				availableServices,
+			)
+			for key := range defaultConsumerResourceNames {
+				consumerConfigMap.Data[key] = cmp.Or(
+					strings.Trim(consumerConfigMap.Data[key], " "),
+					defaultConsumerResourceNames[key],
+				)
+			}
+		}
+
+		annotations := consumerConfigMap.GetAnnotations()
+		primaryConsumerUID, hasAnnotation := annotations[primaryConsumerUIDAnnotation]
+		hasMatchingOwner := false
+		if hasAnnotation {
+			for _, ref := range consumerConfigMap.OwnerReferences {
+				if string(ref.UID) == primaryConsumerUID {
+					hasMatchingOwner = true
+					break
+				}
+			}
+		}
+		if !hasAnnotation || !hasMatchingOwner {
+			util.AddAnnotation(consumerConfigMap, primaryConsumerUIDAnnotation, string(r.storageConsumer.UID))
+		}
```

Comment:
This entire code section can lose a lot of assumption and can be collapsed into:

```suggestion
		primaryConsumerUID := consumerConfigMap.GetAnnotations()[primaryConsumerUIDAnnotation]
		primaryRefIndex := slices.IndexFunc(
			consumerConfigMap.OwnerReferences,
			func(ref metav1.OwnerReference) bool { return ref.UID == primaryConsumerUID) },
		)
		if  primaryRefIndex == -1 {
			util.AddAnnotation(consumerConfigMap, primaryConsumerUIDAnnotation, string(r.storageConsumer.UID))
		}
```